server/src/main/java/keywhiz/commands/DropDeletedSecretsCommand.java

```diff
@@ -0,0 +1,129 @@
+package keywhiz.commands;
+
+import io.dropwizard.cli.ConfiguredCommand;
+import io.dropwizard.setup.Bootstrap;
+import java.io.Console;
+import java.util.Scanner;
+import javax.inject.Inject;
+import keywhiz.KeywhizConfig;
+import keywhiz.service.daos.SecretDAO;
+import net.sourceforge.argparse4j.inf.Namespace;
+import net.sourceforge.argparse4j.inf.Subparser;
+import org.joda.time.DateTime;
+
+import static java.lang.String.format;
+
+/**
+ * Secrets which are deleted from Keywhiz--whether through the admin client or the automation
+ * interface--are not actually removed from Keywhiz' database.  Instead, the link between the
+ * "secret series" and "secret contents" tables is broken, but it is still possible to reconnect
+ * them manually and restore a deleted secret.
+ *
+ * This command permanently drops information about deleted secrets from the database.  This is
+ * clearly dangerous, so it should be run only when it is certain that the deleted secrets will not
```

Comment:
One possible option to make this less dangerous would be to put a minimum threshold into the code, so no matter what date you enter, it will never drop things that haven't been deleted for at least a week (or a month or whatever, just an arbitrary number for the example).

Reply:
True; it seems, though, like what threshold is safe is going to depend fairly heavily on how Keywhiz is being used (and setting too strict of a threshold will make this endpoint less useful).

Reply:
That's definitely true that it depends on how you use keywhiz. What if it was a configurable value?

Reply:
That could work; I'm not convinced that it adds a lot of protection beyond what's already provided by the configurable `deletedBefore` argument and the confirmation prompts, though.